### DISCUSSION TITLE
Getting error when installing pan-os-python

### DISCUSSION THREAD

**Discussion by mohitm84** [PaloAltoNetworks/pan-os-python#295] [Q&A] [answered]
Hello everyone,

I am quite new to python and coding too and I am stuck with below errors when attempting to install the pan-os-python package on my office PC with Win10. Can someone help me in fixing this. 
Thanks in advance.

==========
C:\Users\mm>**pip install --upgrade pan-os-python**
Collecting pan-os-python
  Retrying (Retry(total=4, connect=None, read=None, redirect=None)) after connection broken by 'ProtocolError('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))': /simple/pan-os-python/
  Retrying (Retry(total=3, connect=None, read=None, redirect=None)) after connection broken by 'ProtocolError('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))': /simple/pan-os-python/
  Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'ProtocolError('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))': /simple/pan-os-python/
  Retrying (Retry(total=1, connect=None, read=None, redirect=None)) after connection broken by 'ProtocolError('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))': /simple/pan-os-python/
  Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'ProtocolError('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))': /simple/pan-os-python/
  Could not find a version that satisfies the requirement pan-os-python (from versions: )
No matching distribution found for pan-os-python

**C:\Users\mm>python**
**Python 3.6.0 (v3.6.0:41df79263a11, Dec 23 2016, 08:06:12) [MSC v.1900 64 bit (AMD64)] on win32**
Type "help", "copyright", "credits" or "license" for more information.
>>> ^Z

**C:\Users\mm>pip --version**
**pip 9.0.1 from c:\python36\lib\site-packages (python 3.6)**

C:\Users\mm>
==========

**Comment by btorresgil** [comment, marked as answer]
Are you able to install other packages with pip? Try `pip install requests` and see if that works?

From the errors, this seems like an issue with the connection from your computer to the server. I recommend to check if there are any proxy servers, especially TLS proxy on your network, or any other firewall that might be blocking this connection.

BTW, thanks for using the new GitHub Discussions for this question!

**Reply by mohitm84** [reply]
Thanks for quick response Brian. I just ran pip install requests and it gave me exactly same errors. I am noticing drops on firewall from my source ip to destination 208.67.x.x on port 443. I am not sure but I think that this dropped traffic is related to these pip commands I am running. Are you able to suggest if pip install triggers tcp-443 traffic to destination 208.67.x.x ?
Thanks.

**Reply by KedharnathGoud** [reply]
Your office won't allow you to access Internet directly it should go through a  proxy server. Its a proxy issue you need add  - - proxy and its address to pip.  please find the link for your reference. 
you can create proxy by your self or ask the  experienced python developer in your office which server he is using or you can get the proxy servers list in a pac file which will be under your Internet explorer lan settings

https://pip.pypa.io/en/stable/user_guide/

**Reply by mohitm84** [reply]
Thanks KedarnathGoud and btorresgil. I was being very stupid and didnot attempt the installation from my personal internet, now I just tryied installing it from my personal internet and it just worked smoothly. I also tested a python script successfully also. Thanks once again for your help.
